core/src/main/java/org/elasticsearch/bootstrap/Security.java

```diff
@@ -284,19 +286,60 @@ static void addBindPermissions(Permissions policy, Settings settings) throws IOE
         for (Map.Entry<String, Settings> entry : profiles.entrySet()) {
             Settings profileSettings = entry.getValue();
             String name = entry.getKey();
-            String transportRange = profileSettings.get("port", TransportSettings.PORT.get(settings));
 
             // a profile is only valid if its the default profile, or if it has an actual name and specifies a port
             boolean valid = TransportSettings.DEFAULT_PROFILE.equals(name) || (Strings.hasLength(name) && profileSettings.get("port") != null);
             if (valid) {
-                // listen is always called with 'localhost' but use wildcard to be sure, no name service is consulted.
-                // see SocketPermission implies() code
-                policy.add(new SocketPermission("*:" + transportRange, "listen,resolve"));
+                addSocketPermissionForTransport(policy, profileSettings);
+            }
+        }
+
+        final Map<String, Settings> tribeNodesSettings = new HashMap<>(settings.getGroups("tribe", true));
+        for (final Settings tribeNodeSettings : tribeNodesSettings.values()) {
+            // tribe nodes have HTTP disabled by default, so we check if HTTP is enabled before granting
+            if (NetworkModule.HTTP_ENABLED.exists(tribeNodeSettings) && NetworkModule.HTTP_ENABLED.get(tribeNodeSettings)) {
```

Comment:
Yeah, I'm not sure why we allow HTTP to be enabled on the internal node clients; at least it's disabled by default. :smile: